docs/source/manual/di.rst

```diff
@@ -0,0 +1,158 @@
+.. _man-di:
+
+################################
+Dropwizard Dependency Injection
+################################
+
+.. highlight:: text
+
+.. rubric:: Dropwizard provides you with simple dependency injection mechanism, using HK2,
+            out-of-the-box, and you can add support for more advanced DI by using Guice bundle.
```

Comment:
Would be nice if "Guice Bundle" had some hyperlink. 

Or also mention Dagger2, if people are using that (referenced in the linked issue)

Reply:
@OneCricketeer Could you create a pull request for this?

Reply:
Sure. One question, are we recommending <https://github.com/xvik/dropwizard-guicey> ? I personally haven't used that one, but our DW 1.x projects are [using the Hubspot one, but the maintainer said it was abandoned](https://github.com/HubSpot/dropwizard-guicier/issues/61#issuecomment-586598066).

Also not really aware of a public bundle that actually uses Dagger w/ DW